pylabrobot/centrifuge/agilent_centrifuge.py

```diff
@@ -0,0 +1,399 @@
+      await self.send(b"\xaa\x02\x0e\x10")
+    await self.send(b"\xaa\x02\x0e\x10")
+    await self.lock_door()
+
+    await self.send(b"\xaa\x01\x0e\x0f")
+    await self.send(b"\xaa\x02\x0e\x10")
+
+    await self.send(b"\xaa\x01\x0e\x0f")
+    await self.send(b"\xaa\x02\x0e\x10")
+
+    await self.send(b"\xaa\x01\x0e\x0f")
+    await self.send(b"\xaa\x02\x0e\x10")
+
+    await self.send(b"\xaa\x02\x0e\x10")
+    await self.send(b"\xaa\x01\x0e\x0f")
+
+    await self.send(b"\xaa\x02\x0e\x10")
+    await self.send(b"\xaa\x02\x26\x00\x00\x28")
+    await self.send(b"\xaa\x02\x0e\x10")
+
+    await self.send(b"\xaa\x02\x0e\x10")
+    await self.send(b"\xaa\x01\x0e\x0f")
+    await self.send(b"\xaa\x02\x0e\x10")
+
+    await self.send(b"\xaa\x01\x17\x02\x1a")
+    await self.send(b"\xaa\x01\x0e\x0f")
+    await self.send(b"\xaa\x01\xe6\xc8\x00\xb0\x04\x96\x00\x0f\x00\x4b\x00\xa0\x0f\x05\x00\x07")
+    await self.send(b"\xaa\x01\x17\x04\x1c")
+    await self.send(b"\xaa\x01\x17\x01\x19")
+
+    await self.send(b"\xaa\x01\x0b\x0c")
+    await self.send(b"\xaa\x01\x00\x01")
+    await self.send(b"\xaa\x01\xe6\x05\x00\x64\x00\x00\x00\x00\x00\x32\x00\xe8\x03\x01\x00\x6e")
+    await self.send(b"\xaa\x01\x94\xb6\x12\x83\x00\x00\x12\x01\x00\x00\xf3")
+    await self.send(b"\xaa\x01\x19\x28\x42")
+    await self.send(b"\xaa\x01\x0e\x0f")
+
+    resp = "89"
+    while resp == "89":
+      await self.send(b"\xaa\x02\x0e\x10")
+      resp = await self.send(b"\xaa\x01\x0e\x0f")
+      resp = f"{resp[0]:02x}"
+
+    await self.send(b"\xaa\x01\x0e\x0f")
+    await self.send(b"\xaa\x01\x0e\x0f")
+
+    await self.send(b"\xaa\x01\x17\x02\x1a")
+    await self.send(b"\xaa\x01\x0e\x0f")
+    await self.send(b"\xaa\x01\xe6\xc8\x00\xb0\x04\x96\x00\x0f\x00\x4b\x00\xa0\x0f\x05\x00\x07")
+    await self.send(b"\xaa\x01\x17\x04\x1c")
+    await self.send(b"\xaa\x01\x17\x01\x19")
+
+    await self.send(b"\xaa\x01\x0b\x0c")
+    await self.send(b"\xaa\x01\x0e\x0f")
+    await self.send(b"\xaa\x01\xe6\xc8\x00\xb0\x04\x96\x00\x0f\x00\x4b\x00\xa0\x0f\x05\x00\x07")
+    new_postiion = (self.homing_position + 8000).to_bytes(4, byteorder="little")
+    await self.send(b"\xaa\x01\xd4\x97" + new_postiion + b"\xc3\xf5\x28\x00\xd7\x1a\x00\x00\x49")
+    await self.send(b"\xaa\x01\x0e\x0f")
+    await self.send(b"\xaa\x01\x0e\x0f")
+
+    resp = "08"
+    while resp != "09":
+      resp = await self.send(b"\xaa\x01\x0e\x0f")
+      await self.send(b"\xaa\x01\x0e\x0f")
+      resp = f"{resp[0]:02x}"
+    await self.send(b"\xaa\x01\x0e\x0f")
+    await self.send(b"\xaa\x01\x0e\x0f")
+
+    await self.send(b"\xaa\x01\x17\x02\x1a")
+
+    await self.send(b"\xaa\x02\x0e\x10")
+    await self.lock_door()
+
+    await self.send(b"\xaa\x01\x0e\x0f")
+
+  async def stop(self):
+    await self.send(b"\xaa\x02\x0e\x10")
+    await self.configure_and_initialize()
+    if self.dev:
+      self.dev.close()
+
+  async def get_status(self):
+    resp = await self.send(b"\xaa\x01\x0e\x0f")
```

Comment:
```suggestion
    resp = await self.send(b"\xaa\x01\x0e\x0f")
    if len(resp) == 0:
        raise IOError("Empty status from centrifuge")
```

seems like you assume this is non empty below when you set `self.status`